trunk/modules/python/dataset/reader.py

```diff
@@ -5,7 +5,9 @@
 def normalize(v) :
     '''Normalize a vector in a naive manner.'''
     minimum, maximum = np.amin(v), np.amax(v)
-    return (v - minimum) / (maximum - minimum)
+    diff = maximum - minimum
+    denom = diff if diff != 0. else 1.
+    return (v - minimum) / denom
 
```

Comment:
Here there is no reason to perform the operation if diff is 0. In this case just "return v if diff == 0. else " perform operation. I'm not sure how python handles this, but we should also check within an epsilon.